interface/admin.py

```diff
@@ -1,6 +1,48 @@
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
 from django.contrib import admin
+from django.db.models import F
+from django.http import HttpResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, _assignment):
+        assignment = _assignment[0]
+
+        subs = Submission.objects.filter(
+                assignment=assignment).order_by(F('timestamp').asc())
+
+        submissions = {}
+
+        for submission in subs:
+            submissions[submission.user.username] = submission
```

Comment:
I think you can add a comment like "we only want the last submission", I was confused at first, until I remembered we talked about this.